src/IJ_mv/aux_parcsr_matrix.c

```diff
@@ -75,33 +77,68 @@ hypre_AuxParCSRMatrixCreate( hypre_AuxParCSRMatrix **aux_matrix,
 HYPRE_Int
 hypre_AuxParCSRMatrixDestroy( hypre_AuxParCSRMatrix *matrix )
 {
-   HYPRE_Int ierr = 0;
-   HYPRE_Int i;
-   HYPRE_Int num_rows;
+   HYPRE_Int   num_rownnz;
+   HYPRE_Int   num_rows;
+   HYPRE_Int  *rownnz;
+   HYPRE_Int   i;
 
    if (matrix)
    {
+      rownnz     = hypre_AuxParCSRMatrixRownnz(matrix);
+      num_rownnz = hypre_AuxParCSRMatrixLocalNumRownnz(matrix);
       num_rows = hypre_AuxParCSRMatrixLocalNumRows(matrix);
-      hypre_TFree(hypre_AuxParCSRMatrixRowLength(matrix), HYPRE_MEMORY_HOST);
-      hypre_TFree(hypre_AuxParCSRMatrixRowSpace(matrix), HYPRE_MEMORY_HOST);
 
       if (hypre_AuxParCSRMatrixAuxJ(matrix))
       {
+         if (hypre_AuxParCSRMatrixRownnz(matrix))
+         {
+            for (i = 0; i < num_rownnz; i++)
+            {
+               hypre_TFree(hypre_AuxParCSRMatrixAuxJ(matrix)[rownnz[i]], HYPRE_MEMORY_HOST);
+            }
+         }
+         else
+         {
+            for (i = 0; i < num_rows; i++)
+            {
+               hypre_TFree(hypre_AuxParCSRMatrixAuxJ(matrix)[i], HYPRE_MEMORY_HOST);
+            }
+         }
+
          for (i = 0; i < num_rows; i++)
```

Comment:
What's the purpose of this for-loop?

Reply:
It's actually unnecessary. I used it for debugging and forgot to delete. Thanks for catching this!